src/components/table.js

```diff
@@ -33,80 +35,129 @@ export class Table extends Component {
       this.state.sortField !== prevState.sortField ||
       this.state.sortUp !== prevState.sortUp
-    )
-      newState.sortedData = this.sortData(this.props.data);
-
-    // if sorted data changes or search string changes
-    // update filtered data
-    if (
-      !compareObjects(this.state.sortedData, prevState.sortedData) ||
-      this.state.searchString !== prevState.searchString
     ) {
+      newState.sortedData = this.sortData(this.state.indexedData);
+      newState.filteredData = this.filterData(newState.sortedData);
+      newState.paginatedData = this.paginateData(newState.filteredData);
+      newState.data = newState.paginatedData;
+    }
+
+    // when search string changes
+    if (this.state.searchString !== prevState.searchString) {
       newState.filteredData = this.filterData(this.state.sortedData);
-      if (this.state.searchString !== prevState.searchString)
-        newState.page = 1;
+      newState.searchResults = newState.filteredData.length || 0;
+      newState.paginatedData = this.paginateData(newState.filteredData);
+      newState.pages = this.calcPages(
+        newState.filteredData,
+        this.state.perPage
+      );
+      newState.page = 1;
+      newState.data = newState.paginatedData;
     }
 
-    // if filtered data changes or page controls change
-    // update paginated data
+    // when page controls change
     if (
-      !compareObjects(this.state.filteredData, prevState.filteredData) ||
       this.state.page !== prevState.page ||
       this.state.perPage !== prevState.perPage
     ) {
       newState.paginatedData = this.paginateData(this.state.filteredData);
-      newState.pages = this.calcPages();
+      newState.pages = this.calcPages(
+        this.state.filteredData,
+        this.state.perPage
+      );
+      newState.data = newState.paginatedData;
     }
 
-    // if paginated data changes
-    // update main final data var that is passed to children for render
-    if (!compareObjects(this.state.paginatedData, prevState.paginatedData))
-      newState.data = this.state.paginatedData;
-
     // set new state, if any
     if (Object.keys(newState).length > 0)
       this.setState(newState);
   }
 
-  // change which field table is sorted by
-  changeSort = (field) => {
-    const newState = {};
-    newState.sortField = field;
+  // //////////////////////////////////////////////////
+  // DATA FUNTIONS
+  // //////////////////////////////////////////////////
 
-    if (field === this.state.sortField)
-      newState.sortUp = !this.state.sortUp;
-    else
-      newState.sortUp = true;
+  // function to change input props.data
+  setData = (data) => {
+    data = copyObject(data);
 
-    this.setState(newState);
+    // remove row index property from each object
+    for (const datum of data)
+      delete datum[rowIndexKey];
+
+    // call onChange function with new data to set
+    const func = this.props.onChange || (() => null);
```

Comment:
So I suppose lines 146-147 are equivalent to the following code?
```JS
if (this.props.onChange) {
  this.props.onChange(data);
}
```
Or more simply:
```JS
this.props.onChange && this.props.onChange(data)
```